integration/mock-backend-server/src/main/java/org/wso2/choreo/connect/mockbackend/http2/EchoHttpServerHandler.java

```diff
@@ -0,0 +1,102 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.mockbackend.http2;
+
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http2.HttpConversionUtil;
+
+import java.util.concurrent.TimeUnit;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+
+/**
+ * Handler implementation for the echo server and http/2 echo server with
+ * content aggregation.
+ * For http/2 echo server with content aggregation, this receives a
+ * {@link FullHttpRequest},
+ * which has been converted by a
+ * {@link io.netty.handler.codec.http2.InboundHttp2ToHttpAdapter} before it
+ * arrives here.
+ * For further details, check {@link Http2OrHttpHandler} where the pipeline is
+ * setup.
+ */
+@Sharable
+public class EchoHttpServerHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
+
+    private long sleepTime;
+    private boolean h2ContentAggregate;
+
+    EchoHttpServerHandler(long sleepTime, boolean h2ContentAggregate) {
+        this.sleepTime = sleepTime;
+        this.h2ContentAggregate = h2ContentAggregate;
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) {
+        if (h2ContentAggregate) {
+            String streamId = request.headers().get(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text());
+            FullHttpResponse response = buildFullHttpResponse(request);
+            response.headers().set(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text(), streamId);
+            ctx.writeAndFlush(response);
+        } else {
+            // Decide whether to close the connection or not
+            boolean keepAlive = HttpUtil.isKeepAlive(request);
+            // Build the response object
+            FullHttpResponse response = buildFullHttpResponse(request);
+            if (keepAlive) {
+                // Add keep alive header
+                response.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.KEEP_ALIVE);
+            }
+            if (sleepTime > 0) {
```

Comment:
When does the sleep time become useful?

Reply:
This handler to handle http1 connection. This was the exact code in https://github.com/wso2
/
[performance-common](https://github.com/wso2/performance-common). I didn't look into that code since the tests are for http2